packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -3504,6 +3084,282 @@ merge_matrices(CrsMatrixStruct<Scalar, LocalOrdinal, GlobalOrdinal, Node>& Aview
   }
 }//end merge_matrices
 
+namespace ColMapFunctors
+{
+  template<typename GO, typename GView, typename bitset_t>
+  struct GatherPresentEntries
+  {
+    GatherPresentEntries(GO minGID_, const GView& gids_, bitset_t& present_)
+      : minGID(minGID_), gids(gids_), present(present_)
+    {}
+
+    KOKKOS_INLINE_FUNCTION void operator()(const GO i) const
+    {
+      present.set(gids(i) - minGID);
+    }
+
+    GO minGID;
+    GView gids;
+    bitset_t present;
+  };
+
+  template<typename GO, typename GView, typename const_bitset_t>
+  struct ListGIDs
+  {
+    ListGIDs(GO minGID_, GView& gidList_, const_bitset_t& present_)
+      : minGID(minGID_), gidList(gidList_), present(present_)
+    {}
+
+    KOKKOS_INLINE_FUNCTION void operator()(const GO i, GO& lcount, const bool& finalPass) const
```

Comment:
Pass `bool` input parameters by value, not by reference.  This code is correct, but you're taking up `sizeof(void*)` on the stack rather than `sizeof(bool)`.